.github/workflows/pr-realm-js.yml

```diff
@@ -0,0 +1,391 @@
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          submodules: "recursive"
+
+      - name: Setup node version
+        uses: actions/setup-node@v2
+        with:
+          node-version: '14'
+          cache: 'npm'
+          cache-dependency-path: '**/package-lock.json'
+
+      - name: Setup NPM version
+        run: npm install -g npm@7
+
+      - name: Get NPM cache directory
+        id: npm-cache-dir
+        run: echo "::set-output name=dir::$(npm config get cache)"
+
+      - name: Restore NPM cache
+        id: npm-cache
+        uses: actions/cache@v2
+        with:
+          path: ${{ steps.npm-cache-dir.outputs.dir }}
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-
+
+      - name: Cache prebuild artifacts
+        id: cache-prebuild
+        uses: actions/cache@v2
+        env:
+          cache-name: prebuild
+        with:
+          path: ${{ matrix.variant.artifact-path }}
+          key: ${{ runner.os }}-${{ matrix.variant.os }}-${{ matrix.variant.arch }}-${{ env.cache-name }}-${{hashFiles('./src/**', './vendor/**')}}
+
+      - name: MSVC Setup
+        if: ${{ runner.os == 'Windows' }}
+        uses: ilammy/msvc-dev-cmd@v1
+
+      - name: Setup Vcpkg
+        if: ${{ runner.os == 'Windows' }}
+        uses: friendlyanon/setup-vcpkg@v1
+        with:
+          path: vendor/realm-core/tools/vcpkg/ports
+          cache-key: vcpkg-windows-${{ matrix.variant.arch }}
+          cache-restore-keys: vcpkg-windows-${{ matrix.variant.arch }}
+
+      # ninja-build is used by default if available and results in faster build times
+      # On linux, electron requires a connected display.  We fake this by giving it a headless environment using xvfb
+      # Relevant issue: https://github.com/juliangruber/browser-run/issues/147
+      - name: Linux Environment setup
+        if: ${{ (matrix.variant.runner == 'ubuntu-20.04') }}
+        run: sudo apt-get install ccache ninja-build xvfb
+
+      - name: ccache
+        uses: hendrikmuhs/ccache-action@v1
+        #TODO: figure out using ccache or equivalent on windows
+        if: ${{ runner.os != 'Windows' }}
+        with:
+          key: ${{ runner.os }}-${{ matrix.variant.os }}-${{ matrix.variant.arch }}
+          max-size: '2.0G'
+
+      - name: Prepend ccache executables to the PATH
+        if: ${{ runner.os != 'Windows' }}
+        run: |
+          echo "/usr/lib/ccache:/usr/local/opt/ccache/libexec" >> $GITHUB_PATH
+
+      - name: Configure ccache
+        if: ${{ runner.os != 'Windows' }}
+        run: |
+          ccache --set-config="compiler_check=content"
+          ccache --show-config
+
+      - name: Install deps
+        run: npm ci --ignore-scripts
+
+      # Install node dependencies
+      - name: Install prebuilds
+        if: steps.cache-prebuild.outputs.cache-hit == 'true'
```

Comment:
Not sure I understand this - why do we need to run `npm ci` if the cache was restored?

Reply:
Agreed